x/language.rb

```diff
@@ -0,0 +1,18 @@
+module X
+  class Language
+    def self.of(id)
+      tracks[id.to_s]
+    end
+
+    def self.tracks
+      @@tracks ||= fetch_tracks.inject({}) do |hash, track|
+        hash[track.id] = track.language
+        hash
+      end
+    end
+
+    def self.fetch_tracks
```

Comment:
Can we name this for what it returns, not what it does?

Also, let's not make this a public method.